numpy/_core/src/multiarray/descriptor.c

```diff
@@ -2877,6 +2878,7 @@ arraydescr_setstate(_PyArray_LegacyDescr *self, PyObject *args)
     char endian;
     PyObject *endian_obj;
     PyObject *subarray, *fields, *names = NULL, *metadata=NULL;
+    PyObject *old_metadata, *new_metadata;
```

Comment:
Would probably move these down, but doesn't matter (and yeah, everything else is here so).

Reply:
Yes, they seem more local state - I moved them down to be clearer.